### PR TITLE
ci: test on 32bit os

### PR DESCRIPTION
**Pull request by magik6k**
To test for stuff like https://github.com/ipfs/go-ipfs/issues/5367

There is also one test fail in gateway test code:
```
core/corehttp/gateway_test.go:42:9: constant 18446744073709551615 overflows uint
```

### CODE REVIEW

**Comment by magik6k** [comment]
Blocked on https://github.com/ipfs/go-ipfs/issues/5367 / propagating new badger

File: test/bin/Rules.mk [hunk 1]
```diff
@@ -30,18 +30,18 @@ TGTS_GX_$(d) := hang-fds
 TGTS_GX_$(d) := $(addprefix $(d)/,$(TGTS_GX_$(d)))
 
 $(TGTS_GX_$(d)):
-	go build -i $(go-flags-with-tags) -o "$@" "$(call gx-path,$(notdir $@))"
```
Comment (Stebalien) [inline test/bin/Rules.mk]: Do you remember why we needed to remove this flag?

Comment (magik6k) [reply test/bin/Rules.mk]: IIRC when cross compiling, go would try to write objects to /usr

Comment (Stebalien) [reply test/bin/Rules.mk]: Ah. Yeah, that makes sense.

**Comment by Stebalien** [comment]
Still blocked: https://github.com/ipfs/go-ipfs/issues/5367#issuecomment-457247271

**Pull request merged**